### PR TITLE
Catalyst integration

### PR DESCRIPTION
**Pull request by acbauer**
Hi,

This pull request includes changes to get VPIC instrumented to use ParaView Catalyst for in situ analysis and visualization. I'm not sure what the proper way to get code changes reviewed and submitted into the main VPIC repo is so if there are steps that I missed or changes that need to be made, please let me know -- I'm more than happy to do that to help maintain consistency inside of VPIC. Also, please let me know if you have questions and/or if the commit messages and code comments are lacking.

Thanks,
Andy

### CODE REVIEW

**Comment by junghans** [comment]
@dnystrom1 : Make sure that Kitware has sign the CLA before merging this!

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "cinch"]
 	path = cinch
-	url = ../cinch.git
+	url = https://github.com/losalamos/cinch.git
```
Comment (junghans) [inline .gitmodules]: I would revert this change!

Comment (acbauer) [inline .gitmodules]: Where is the CLA and where do I send it after I get Kitware's representative's signature on the form?

As for the cinch submodule change -- this can cause problems if someone isn't cloning directly from https://github.com/losalamos/vpic.git since this is a relative instead of an absolute path. Try the following from the main repo's master branch and then the branch with my changes:
> git clone https://github.com/losalamos/vpic.git
> cd vpic
> git submodule update --init -r
> cd ..
> mkdir vpic2
> cd vpic2
> git clone ../vpic

**Comment by dnystrom1** [comment]
OK. But what is the CLA? I've been using git for about 10 years 
now but I am new to the process of dealing with pull requests, etc. 
Also, really busy at the moment but will start working on reviewing 
the changes and getting them approved by the VPIC Team. 

Thanks, 

Dave 

----- Original Message -----

File: .gitmodules (hunk 1 above)
Comment (junghans) [inline .gitmodules]: The reason why we use relative paths is that we are mirroring vpic onto machines, which don't have internet access.

**Comment by junghans** [comment]
CLA is an requirement for external contributors, I just made an electronic version here:
https://www.clahub.com/agreements/losalamos/vpic

File: .gitmodules (hunk 1 above)
Comment (acbauer) [inline .gitmodules]: Ok. I'll revert this change.

**Comment by acbauer** [comment]
Note that I've signed the online CLA.

**Comment by junghans** [comment]
I guess @demarle needs to sign too!

**Comment by acbauer** [comment]
FYI: Dave DeMarle has also now signed the CLA, in case you didn't get a notification for that.

**Comment by acbauer** [comment]
Hi, I wanted to ping folks on this again in case the pull request has stalled somewhere. Please let me know if I can help push this along in any way.

**Comment by dnystrom1** [comment]
I just returned from a week of travel for the factory test of the 
second half of the new Trinity computer being installed here 
at LANL. And the time leading up to that and also since 
getting back has been very busy for me. But this pull request 
is on my To Do list and I hope to get to it soon. 

Sorry for the delay, 

Dave 

----- Original Message -----

**Comment by acbauer** [comment]
I'm closing this since Chet Nieter's pull request (#4) adds in extra functionality for dealing with particles as well as rebasing the changes to be off of a more recent version of VPIC.
